### PR TITLE
feat(pie-css): DSW-1040 add font-size function and improve tests

### PR DESCRIPTION
**Pull request by ashleynolan**
## Describe your changes

- [Added] - `font-size` function added & tests
- [Added] - adding pie-css to pie-button

## Author Checklist
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Review by jamieomaguire — DISMISSED** [review]

**Review by jamieomaguire — APPROVED** [review]

File: commitlint.config.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ module.exports = {
             headerPattern: /^(\w*)\((\w.*)\): (DSW-\d{1,7}) (\w.*)$/,
             headerCorrespondence: ['type', 'scope', 'ticket', 'subject'],
             issuePrefixes: [czConfig.ticketNumberPrefix],
+            referenceActions: null, // this stops commitlint erroring when using these words ['close', 'closes', 'closed', 'fix', 'fixes', 'fixed', 'resolve', 'resolves', 'resolved']
```
Comment (raoufswe) [inline commitlint.config.js:16]: 😄

File: packages/tools/pie-css/scss/mixins/_fontSize.scss [hunk 2]
```diff
@@ -11,8 +11,11 @@
...
-// Outputs: .text { font-size: calc(var(--font-size-small) * 1px); }
-// -
+// Outputs:
+// .text {
+//   font-size: calc(var(--font-size-small) * 1px);
+// }
+// ----------------------------------------------
 @mixin font-size($token) {
     font-size: calc(var($token) * 1px);
 }
```
Comment (raoufswe) [inline packages/tools/pie-css/scss/mixins/_fontSize.scss:21]: since both the mixin and function implement the same logic, when should we use each of them?

Comment (ashleynolan) [reply packages/tools/pie-css/scss/mixins/_fontSize.scss:21]: The function is for when we want to use font-size for assignment (like in variables):

```scss
--my-font-size: #{p.font-size(--dt-var)};

// will give you
// --my-font-size: calc(…);
```

The mixin is if you want the output to include the left-hand-side `font-size:` as well:
```scss
@include font-size(--dt-var);

// will give you
// font-size: calc(…);
```

Comment (raoufswe) [reply packages/tools/pie-css/scss/mixins/_fontSize.scss:21]: makes sense 💯

**Review by raoufswe — APPROVED** [review]

**Pull request merged**